From 841ae6ddddf848a05509e27ad516e4d5c1758f10 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Fri, 28 Feb 2020 13:20:23 +0530
Subject: [PATCH] meson: Use libtool library versioning, same as autotools

This helps maintain ABI compatibility with the Autotools build so it's
a drop-in replacement.
---
 meson.build     | 3 +++
 src/meson.build | 1 +
 2 files changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index 773e6af..2f54339 100644
--- a/meson.build
+++ b/meson.build
@@ -2,6 +2,9 @@ project('libpsl', 'c',
   version : '0.20.2',
   meson_version : '>=0.47.0')
 
+# Derived from LIBPSL_SO_VERSION in configure.ac
+lt_version = '5.3.2'
+
 cc = meson.get_compiler('c')
 
 enable_runtime = get_option('runtime')
diff --git a/src/meson.build b/src/meson.build
index 9bce343..50a272e 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -20,6 +20,7 @@ libpsl = library('psl', sources, suffixes_dafsa_h,
   include_directories : [configinc, includedir],
   c_args : cargs,
   dependencies : [libidn2_dep, libidn_dep, libicu_dep, libunistring, networking_deps],
+  version: lt_version,
   install: true,
 )
 
-- 
2.24.1

